unfurl/src/main/kotlin/me/saket/unfurl/extension/HtmlTagsBasedUnfurler.kt

```diff
@@ -33,8 +34,8 @@ open class HtmlTagsBasedUnfurler : UnfurlerExtension {
 
         if (body != null && body.contentType().isHtmlText()) {
           // TODO: stream the HTML body only until a "</head>" is received instead of streaming the entire HTML body.
-          Jsoup.parse(
-            /* in */ body.source().inputStream(),
+          Ksoup.parse(
+            /* in */ BufferReader(body.source().buffer),
```

Comment:
Bad news: `ksoup` doesn't actually implement streaming parsing. It internally allocates the entire content in memory.

https://github.com/fleeksoft/ksoup/blob/7365999c33d9f9ba37d9c3bde64fc63f7eff561a/ksoup/src/commonMain/kotlin/com/fleeksoft/ksoup/ported/BufferReader.kt#L35

We'll unfortunately not be able to use `ksoup` until they fix this problem 😕

Reply:
There may be another option: `unfurl` only needs the `<head>` portion of web pages. Can `unfurl` stream the HTML until a `</head>` is found and then give it to `ksoup`? It'll still allocate everything in memory but at least the allocation will be significantly smaller.

Reply:
I filed an issue here: https://github.com/fleeksoft/ksoup/issues/2

Reply:
Saw that...Lets wait for him to reply.
Handling streaming in the library itself could work, however ideally would want the parser to handle this logic

Reply:
@saket 
Can we make other changes in the meantime?
Such as overhauling the library to a multiplatform one and making a dev release?
That way we will be sure everything else is in place

Reply:
Will it only contain JVM sources then?

Reply:
No, I meant. 
We will continue with ksoup only
and make other changes on a separate branch such as Ktor and source sets
Wait for ksoup to come up with a change, else I will try to submit a PR for them

Reply:
Unfortunately we can't move forward anymore until `ksoup` fixes this. Merging `ksoup` would prevent me from making any releases to `unfurl` until they fix their streaming API.

Reply:
Oh, I mean to continue without merging
Keep this PR updated in any case.
And merge only when Ksoup is on par.
That way you can make newer releases and we can make progress on this as well

Reply:
Long lived branches aren't great so I'd not suggest that. Wanna migrate to ktor only instead?

Reply:
Yes. Lets do this then.
Should I revert Ksoup changes for now?
And use Ktor CIO?

Reply:
@saket I have updated to Ksoup 0.0.5 with streaming support
Can you take a look?

Reply:
@saket Have upgraded to 0.0.6.
Ksoup has migrated to input stream for HTML 
Can you take a look?